Automattic/studio
# Studio: Use appdata file to store last push and pull timestamp

## Related issues

Closes https://github.com/Automattic/dotcom-forge/issues/9987
Closes https://github.com/Automattic/dotcom-forge/issues/10027

## Proposed Changes

This PR stores the data related to the last pull and push times in the `appdata-v1.json` file instead of using local storage. It also ensures that the timestamps are updated when the user uses the pull and push operation multiple times.

## Testing Instructions



* Start Studio app `STUDIO_SITE_SYNC=true npm start`
* On Studio site, navigate to the `Sync` tab
* Connect a WPCOM site (bonus point to also connect a staging site along the way)
* Hover over the `Pull` and `Push` buttons and confirm that you can see `You have not pulled / pushed this site yet.` tooltip over each button
* Open `/Users/{yourUser/Library/Application Support/Studio/appdata-v1.json` and confirm that the connected site has `lastPullTimestamp` and `lastPushTimestamp` are set to `null` on a newly connected site
* Start either a push or pull operation
* Once it is finished, observe that the tooltip changed and now displays how long ago you pushed or pulled a site
* Open `/Users/{yourUser/Library/Application Support/Studio/appdata-v1.json` and confirm that the connected site has `lastPullTimestamp` or `lastPushTimestamp` (depending on which operation you triggered) set to a specific timestamp
* Try repeating a pull or push operation with triggering an error and confirm that the timestamp does NOT get reset if the pull or push is not successful
* Finally, repeat the same operation and confirm that tooltip time gets updated

## Pre-merge Checklist
